### PR TITLE
feature(cache): format freed bytes for readability

### PR DESCRIPTION
**Pull request by Alizter**

### CODE REVIEW

File: bin/cache.ml [hunk 1]
```diff
@@ -35,8 +35,16 @@ let trim ~trimmed_size ~size =
   with
   | Error s -> User_error.raise [ Pp.text s ]
   | Ok { trimmed_bytes } ->
+    let pp_bytes trimmed_bytes =
```
Comment (snowleopard) [inline bin/cache.ml:38]: I think this code should live somewhere in Stdune as it's generally useful.

Comment (snowleopard) [reply bin/cache.ml:38]: (In fact, don't we already have something similar?)

Comment (Alizter) [reply bin/cache.ml:38]: I couldn't find anything about formatting bytes like this in the repo. I can add it to stdune, though I am not sure who else will use it. Where should I include it?

Comment (emillon) [reply bin/cache.ml:38]: There's `bytes_unit` in `Decoder` (`dune_sexp`). You might be able to reuse the table.

Comment (snowleopard) [reply bin/cache.ml:38]: Ha, indeed! Thanks @emillon. I remember seeing something like this -- it even has a comment from me on top :)

Comment (Alizter) [reply bin/cache.ml]: I've factored the parsing and printing of byte units through a library in stdune called Bytes_units. This is backwards compatible and allows aliases for units to easily be added.

Comment (snowleopard) [reply bin/cache.ml]: Thanks!

File: otherlibs/stdune/bytes_unit.ml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      | (_, value) :: ((units', value') :: _ as l) ->
+        if value >= x && x > value' then (List.hd units', value') else loop l
+    in
+    loop @@ List.rev conversion_table
+  in
+  if value = 1L then Printf.sprintf "%Ld %s" x suffix
+  else
+    Printf.sprintf "%.2f %s" (Int64.to_float x /. Int64.to_float value) suffix
+
+let () =
```
Comment (snowleopard) [inline otherlibs/stdune/bytes_unit.ml:26]: It seems weird to test by running code at the top level. Can we turn this into a proper test that runs only as part of the test suite, not on every execution of Dune?

File: otherlibs/stdune/bytes_unit.ml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+let conversion_table =
+  [ ([ "bytes"; "B" ], 1L)
+  ; ([ "KB"; "kB"; "KiB"; "kilobytes" ], 1000L)
```
Comment (snowleopard) [inline otherlibs/stdune/bytes_unit.ml:3]: `KiB` is not 1000, it's 1024. Using `MiB` and `GiB` below isn't accurate either.

**Review by snowleopard — CHANGES_REQUESTED** [review]
Great job with the comments! I left a couple of suggestions.

By the way, it seems weird that no test changed in the process. I guess the reason is that the only test we have continues to use "bytes". I guess it's fine.

File: otherlibs/stdune/bytes_unit.ml (hunk 2 above)
Comment (Alizter) [reply otherlibs/stdune/bytes_unit.ml]: I've moved it into an expect test. Not sure what I was thinking when I wrote it.

**Comment by Alizter** [comment]
I was careful not to change the tests before, but now I have changed it from `bytes` to `B` to be consistent. I've added an expect test making sure the edge cases are handled correctly.

I am not going to add support for MiB, I actually never knew about MiB before and it was an extremely confusing read. If I add it in our suggested units helper messages will also display them etc. so it looks too tricky to be worth it.

File: test/blackbox-tests/test-cases/dune-cache/trim.t [hunk 4]
```diff
@@ -34,7 +34,7 @@ Function to reset build tree and cache.
 Check that trimming does not crash when the cache directory does not exist.
 
   $ dune cache trim --size 0B
-  Freed 0 bytes
+  Freed 0 B
```
Comment (snowleopard) [inline test/blackbox-tests/test-cases/dune-cache/trim.t:37]: The space between 0 and B seems unnecessary.

Comment (snowleopard) [reply test/blackbox-tests/test-cases/dune-cache/trim.t:37]: (We usually write just "5GB".)

Comment (Alizter) [reply test/blackbox-tests/test-cases/dune-cache/trim.t]: I've removed the space.

**Review by snowleopard — APPROVED** [review]
Looks great, thank you!

**Comment by snowleopard** [comment]
> I am not going to add support for MiB, I actually never knew about MiB before and it was an extremely confusing read. If I add it in our suggested units helper messages will also display them etc. so it looks too tricky to be worth it.

Yeah, I'm also not sure it's worth it.

**Pull request merged**